airsonic-main/src/main/java/org/airsonic/player/dao/LegacyHsqlDaoHelper.java

```diff
@@ -31,29 +51,226 @@ public void checkpoint() {
         LOG.debug("Database checkpoint complete.");
     }
 
-    @PreDestroy
-    public void onDestroy() {
-        Connection conn = null;
+    /**
+     * Return the current version of the HSQLDB database, as reported by the database properties file.
+     */
+    private String getHsqldbDatabaseVersion() {
+        Properties prop = new Properties();
+        File configFile = new File(settingsService.getDefaultJDBCPath() + ".properties");
+        try {
```

Comment:
Please use a try-with-resource construction here ♥

Reply:
Ah, you're entirely right, thanks for noticing!